data/ai/lua/ca_recruit_rushers.lua

```diff
@@ -18,14 +18,15 @@ local params = { score_function = (function() return 196000 end) }
 if ca_castle_switch then
     params.min_turn_1_recruit = (function() return ca_castle_switch:evaluation({}, dummy_engine.data) > 0 end)
     params.leader_takes_village = (function()
-            if ca_castle_switch:evaluation({}, dummy_engine.data) > 0 then
+            local castle_switch_score = ca_castle_switch:evaluation({}, dummy_engine.data)
+            if castle_switch_score > 0 then
                 local take_village = #(wesnoth.get_villages {
                     x = dummy_engine.data.leader_target[1],
                     y = dummy_engine.data.leader_target[2]
                 }) > 0
-                return take_village
+                return castle_switch_score, take_village
             end
-            return not ai.aspects.passive_leader
+            return 0
```

Comment:
Should this be calling your `is_passive_leader` helper?

Reply:
No, because this function has no knowledge of which leader we are talking about (which previously was not required because the aspect applied for the entire side).  So that logic has been moved to generic_recruit_engine.lua.

But this touches on a bigger problem.  If we want to use the Experimental AI CAs for the default AI, there's some serious problems with multi-leader sides.  This all needs to be cleaned up then anyway.

Reply:
It could at least return true if passive_leader is set to yes though, which would maintain backwards compatibility.